stan/math/rev/fun/sum.hpp

```diff
@@ -98,6 +98,22 @@ inline var sum(const EigMat& m) {
   return var(new sum_eigen_v_vari(m_ref));
 }
 
+/**
+ * Returns the sum of the coefficients of the specified
+ * `var_value<matrix/column_vector/row vector>`.
+ *
+ * @tparam T type of the matrix of vector (Must be derived from \c
+ * Eigen::MatrixBase and contain \c var scalars)
+ * @param x Specified var_value containing a matrix or vector.
+ * @return Sum of coefficients of matrix.
+ */
+template <typename T, require_eigen_t<T>* = nullptr>
+inline var sum(const var_value<T>& x) {
```

Comment:
Can you make this work for `Eigen::Matrix<var, R, C>` too? Also convert the `std::vector` specialization?

Reply:
I couldn't figure out the how to get this to work for std::vector as well since this can also take in a regular matrix. Though I shrunk it down to work on `var<mat>` and `mat<var>`

Reply:
The trick is `as_array_or_scalar`. Check out the variance implementation [here](https://github.com/stan-dev/math/blob/62dfc61752298a5ebc2a5b3a7e74e6eecb40f086/stan/math/rev/fun/variance.hpp#L24).

It uses this: https://mc-stan.org/math/dc/d25/as__array__or__scalar_8hpp_source.html

Reply:
I guess `var<mat>` might not work with `as_array_or_scalar`/`as_column_vector_or_scalar` yet. But those should be updated to support `var<mat>` at some point cuz they're super handy functions.

Reply:
Let's do that in another PR. I'd like to have this up since once we have at least a function that uses `var<mat>` and #2024 goes in I can add the indexing stuff for `var<mat>` and we can start doing the work to have `var<mat>` used in the compiler

Reply:
Ofc, ofc, makes sense to do that separately